server/routes/chat_routes.ts

```diff
@@ -314,42 +314,39 @@ export function registerChatRoutes(router: IRouter, routeOptions: RoutesOptions)
       request,
       response
     ): Promise<IOpenSearchDashboardsResponse<HttpResponsePayload | ResponseError>> => {
-      const { sessionId, rootAgentId } = request.body;
+      const { sessionId: sessionIdInRequestBody, rootAgentId, interactionId } = request.body;
       const storageService = createStorageService(context);
-      let messages: IMessage[] = [];
       const chatService = createChatService();
 
+      let outputs: Awaited<ReturnType<ChatService['regenerate']>> | undefined;
+
+      /**
+       * Get final answer from Agent framework
+       */
       try {
-        const session = await storageService.getSession(sessionId);
-        messages.push(...session.messages);
+        outputs = await chatService.regenerate(
+          { sessionId: sessionIdInRequestBody, rootAgentId, interactionId },
+          context
+        );
       } catch (error) {
-        return response.custom({ statusCode: error.statusCode || 500, body: error.message });
+        context.assistant_plugin.logger.error(error);
       }
 
-      const lastInputIndex = messages.findLastIndex((msg) => msg.type === 'input');
-      // Find last input message
-      const input = messages[lastInputIndex] as IInput;
-      // Take the messages before last input message as memory as regenerate will exclude the last outputs
-      messages = messages.slice(0, lastInputIndex);
-
+      /**
+       * Retrieve latest interactions from memory
+       */
+      const sessionId = sessionIdInRequestBody;
```

Comment:
can we rename `sessionIdInRequestBody` to `sessionId`?

Reply:
Sure, done for that.